src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -0,0 +1,491 @@
+                // to use XmlConvert in all other cases, except infinity
+                await writer.WriteAsync(XmlConvert.ToString(value)).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Write a short value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Short value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, short value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a long value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Long value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, long value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteAsync(value.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a double value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Double value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, double value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            if (JsonSharedUtils.IsDoubleValueSerializedAsString(value))
+            {
+                await writer.WriteQuotedAsync(value.ToString(ODataNumberFormatInfo)).ConfigureAwait(false);
+            }
+            else
+            {
+                // double.ToString() supports a max scale of 14,
+                // whereas double.MinValue and double.MaxValue have 16 digits scale. Hence we need
+                // to use XmlConvert in all other cases, except infinity
+                string valueToWrite = XmlConvert.ToString(value);
+
+                await writer.WriteAsync(valueToWrite).ConfigureAwait(false);
+                if (valueToWrite.IndexOfAny(DoubleIndicatingCharacters) < 0)
+                {
+                    await writer.WriteAsync(".0").ConfigureAwait(false);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Write a Guid value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Guid value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, Guid value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteQuotedAsync(value.ToString()).ConfigureAwait(false);

```

Comment:
Add CultureInfo

Reply:
@KenitoInc
The `.ToString()` overloads available when converting a `Guid` to string are as follows:
- `.ToString()`
- `.ToString(string format)`
- `.ToString(string format, IFormatProvider provider)`

We're using the 1st overload and it suffices for a `Guid`